lib/index.js

```diff
@@ -808,6 +810,58 @@ GA.prototype.productClickedEnhanced = function(track) {
 };
 
 /**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
```

Comment:
Before making the impression, we need to verify that either id (which can default to SKU) or name are set - otherwise GA will reject the impression.